pkg/serverless/invocationlifecycle/trace.go

```diff
@@ -0,0 +1,68 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package invocationlifecycle
+
+import (
+	"os"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/trace/api"
+	"github.com/DataDog/datadog-agent/pkg/trace/info"
+	"github.com/DataDog/datadog-agent/pkg/trace/pb"
+)
+
+const (
+	functionNameEnvVar = "AWS_LAMBDA_FUNCTION_NAME"
+	serviceEnvVar      = "DD_SERVICE"
+)
+
+// executionStartInfo is saved information from when an execution span was started
+type executionStartInfo struct {
+	startTime time.Time
+	traceID   uint64
+	spanID    uint64
+}
+
+// currentExecutionInfo represents information from the start of the current execution span
+var currentExecutionInfo executionStartInfo
+
+// startExecutionSpan records information from the start of the invocation.
+// It should be called at the start of the invocation.
+func startExecutionSpan(startTime time.Time) {
+	currentExecutionInfo.startTime = startTime
+	currentExecutionInfo.traceID = random.Uint64()
+	currentExecutionInfo.spanID = random.Uint64()
+}
+
+// endExecutionSpan builds the function execution span and sends it to the intake.
+// It should be called at the end of the invocation.
+func endExecutionSpan(processTrace func(p *api.Payload), endTime time.Time) {
+	duration := endTime.UnixNano() - currentExecutionInfo.startTime.UnixNano()
+
+	executionSpan := &pb.Span{
+		Service:  os.Getenv(serviceEnvVar),
```

Comment:
I think we should get this value for global tags from the daemon as service could be also set in DD_TAGS

Reply:
Could you elaborate on this a bit more? I don’t see a way to retrieve this from the daemon per se — will the trace agent just automatically apply the tag if it is configured?

Reply:
actually you can just set it to aws.lambda and the span processor (which has access to tags, will set the correct value)
https://github.com/DataDog/datadog-agent/blob/main/pkg/serverless/trace/span_modifier.go#L18